lib/segment/src/index/field_index/map_index/mod.rs

```diff
@@ -432,14 +447,23 @@ impl PayloadFieldIndex for MapIndex<IntPayloadType> {
             Some(Match::Value(MatchValue {
                 value: ValueVariants::Integer(integer),
             })) => Ok(self.get_iterator(integer)),
-            Some(Match::Any(MatchAny {
-                any: AnyVariants::Integers(integers),
-            })) => Ok(Box::new(
-                integers
-                    .iter()
-                    .flat_map(|integer| self.get_iterator(integer))
-                    .unique(),
-            )),
+            Some(Match::Any(MatchAny { any: any_variant })) => match any_variant {
+                AnyVariants::Keywords(keywords) => {
+                    if keywords.is_empty() {
+                        Ok(Box::new(vec![].into_iter()))
+                    } else {
+                        Err(OperationError::service_error(
```

Comment:
Why is this a service error and not a client error?

Reply:
tbh, I think it should be something like Option instead of Result, but that's out of scope for this PR